### PR TITLE
Some helper methods

### PR DESCRIPTION
**Pull request by MythTheWolf**
+ Added some helper methods for permissions
+ Implemented a setOverwrittenPermissions method
* Up-to-date with you

Take what you want :ok_hand:

### CODE REVIEW

**Comment by BtoBastian** [comment]
Hey,

thank you very much, but there is some stuff which needs to be changed:
- Please don't use tabs instead of spaces. 
- Please make a seperate PR for eveything you changed. E.g. one PR for the `canXyz()` methods (ofc not every single one) and an other one for the RoleBuilder, etc. 
- Keep the style/format of the other JavaDoc comments
- Make sure you only commit stuff you really want to commit. E.g. your commit also contains a test class with your token... (pro tipp: change it soon! 😉 )
- Some other stuff, but it's hard for me to see all changes you made, because the conversion from tabs to spaces makes it impossible to see single changes
- Use some proper commit messages. `Helper methods stuff`, `More methods` and `Implemented a role builder.` aren't really good commit names for a public library.
- I'm not sure myself, how update methods (like the `setOverwrittenPermissions`) will be done in Javacord 3. It will most likely use a builder pattern, instead of simple setters. Before I'm sure how it will be designed, I won't accept PRs containing methods to update objects.

I'll close this PR for the reasons mentioned above. Feel free to create a new one were the mentioned stuff is fixed, but I would rather prefer to collect the methods (especially the `canXyz()` methods) on Discord and commit them myself afterwards to make sure everything is convenient with the rest of the library.

File: src/main/java/de/btobastian/javacord/entities/permissions/MyFirstBot.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package de.btobastian.javacord.entities.permissions;
+
+import de.btobastian.javacord.DiscordApiBuilder;
+import de.btobastian.javacord.Javacord;
+
+public class MyFirstBot {
+
+    public static void main(String[] args) {
+        String token = "MzU1ODcyOTQ2NjUwMDg3NDI2.DLlx8A.Bjbzq09A-rVyei98FJgfkVNo2zM";
```
Comment (BtoBastian) [inline src/main/java/de/btobastian/javacord/entities/permissions/MyFirstBot.java]: 👍

**Comment by MythTheWolf** [comment]
Ill work on fixes. As far as "collecting on discord",you mean have the suggestions and write them yourself,or..?

**Comment by BtoBastian** [comment]
I'll create a new channel, were everyone can post the implementations for `canXyz()` methods which aren't implemented yet. You'll see what I mean in a few seconds ;)

**Pull request closed without merging**